### PR TITLE
LibCrypto+LibJS: Better bigint bitwise binops

### PR DESCRIPTION
**Pull request by nico**
bitwise_foo() didn't do the right thing for negative numbers. At least it didn't have JS BigInt semantics, and LibJS is currently the only client of these methods.

Fixes test262/test/language/expressions/bitwise-*/bigint.js.

### CODE REVIEW

**Comment by nico** [comment]
Huh, apparently `test-crypto -t bigint` duplicates a bunch of the tests in TestCrypto. What do you want me to do about this? Update the expectations in test-crypto too? Remove the redundant tests in test-crypto?

**Comment by nico** [comment]
Updated the redundant test for now.

**Comment by alimpfard** [comment]
test-crypto isn't used for tests anymore, just recreational usage
We can entirely remove it too, I don't think anyone needs it anymore

File: Userland/Libraries/LibCrypto/BigInt/SignedBigInteger.cpp [hunk 1]
```diff
@@ -194,12 +194,36 @@ FLATTEN SignedBigInteger SignedBigInteger::bitwise_or(const SignedBigInteger& ot
...
+    //   -A & -B == ~(A - 1) & ~(B - 1)
+    // Inverting both sides:
+    //   ~(-A & -B) == ~(~(A - 1) & ~(B - 1)) == ~~(A - 1) | ~~(B - 1) == (A - 1) | (B - 1).
+    // Applying (q1) on the LHS:
+    //   -(-A & -B) - 1 == (A - 1) | (B - 1)
+    // Adding 1 on both sides and then multiplying both sides by -1:
+    //   -A & -B == -( (A - 1) | (B - 1) + 1)
+    // So we can compute the bitwise and by returning a negative number with magnitude (A - 1) | (B - 1) + 1.
+    // This is better than the naive (~A + 1) & (~B + 1) because it needs just one O(n) scan for the or instead of 2 for the ~s.
+    return { unsigned_value().minus(1).bitwise_or(other.unsigned_value().minus(1)).plus(1), true };
```
Comment (alimpfard) [inline Userland/Libraries/LibCrypto/BigInt/SignedBigInteger.cpp:229]: Gosh jolly those minus/plus thingies...
Can't think of any other way though, so 🤷

File: Userland/Libraries/LibCrypto/BigInt/Algorithms/UnsignedBigIntegerAlgorithms.h [hunk 2]
```diff
@@ -18,7 +18,7 @@ class UnsignedBigIntegerAlgorithms {
     static void bitwise_or_without_allocation(UnsignedBigInteger const& left, UnsignedBigInteger const& right, UnsignedBigInteger& output);
     static void bitwise_and_without_allocation(UnsignedBigInteger const& left, UnsignedBigInteger const& right, UnsignedBigInteger& output);
     static void bitwise_xor_without_allocation(UnsignedBigInteger const& left, UnsignedBigInteger const& right, UnsignedBigInteger& output);
-    static void bitwise_not_fill_to_size_without_allocation(UnsignedBigInteger const& left, size_t, UnsignedBigInteger& output);
+    static void bitwise_not_fill_to_one_based_index_without_allocation(UnsignedBigInteger const& left, size_t, UnsignedBigInteger& output);
```
Comment (alimpfard) [inline Userland/Libraries/LibCrypto/BigInt/Algorithms/UnsignedBigIntegerAlgorithms.h:21]: :thonk: can this live in the previous commit?
We can keep the bitwise_or changes here, but seems silly to add something then immediately change it...

**Review by alimpfard — COMMENTED** [review]
Code LGTM (hopefully I didn't miss anything when phone-reviewing), just a few smol comments

**Comment by nico** [comment]
> test-crypto isn't used for tests anymore, just recreational usage

Well, the CI seems to run it :)

**Comment by alimpfard** [comment]
:yakfused: whaaaa? I remember switching to running the LibTest-based one...
Well anyway, feel free to keep it like this or remove it

File: Userland/Libraries/LibCrypto/BigInt/Algorithms/UnsignedBigIntegerAlgorithms.h (hunk 2 above)
Comment (alimpfard) [reply Userland/Libraries/LibCrypto/BigInt/Algorithms/UnsignedBigIntegerAlgorithms.h:21]: Alright, works for me.

**Comment by nico** [comment]
Keeping it like this for now. Might attempt to remove test-crypto later :)

**Pull request merged**